soc/raspberrypi/rpi_pico/rp2350/linker.ld

```diff
@@ -6,6 +6,12 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/*
+ * The Image Definition Block is only linked to the app that
+ * resides at 0x100. This can be the application, or a bootloader
+ * such as mcuboot.
+ */
+#if CONFIG_RP2_REQUIRES_IMAGE_DEFINITION_BLOCK
```

Comment:
Please consider taking a look as #91200 has now landed, and it reorders some of the configs in a way that could benefit your needs